src/openqaoa-core/algorithms/qaoa/qaoa_workflow.py

```diff
@@ -249,11 +250,15 @@ def compile(
             seed=self.circuit_properties.seed,
             total_annealing_time=self.circuit_properties.annealing_time,
         )
-
+        
+        backend_dict = self.backend_properties.__dict__.copy()
+        
         self.backend = get_qaoa_backend(
             qaoa_descriptor=self.qaoa_descriptor,
             device=self.device,
-            **self.backend_properties.__dict__,
+            wrapper=SPAMTwirlingWrapper if self.error_mitigation_properties.error_mitigation_technique == 'spam_twirling' else None,
```

Comment:
A suggestion to consider if it is possible. I was looking into the update you made into `get_qaoa_backend`. Let's see if the following approach would work because this might be slightly better code-wise.
```suggestion
self.backend = get_qaoa_backend(
            qaoa_descriptor=self.qaoa_descriptor,
            device=self.device,
            **backed_dict)
if self.error_mitigation_properties.error_mitigation_technique == 'spam_twirling':
     self.backend = SPAMTwirling(self.backend, **wrapper_options)
```

Reply:
I've edited this now, feel free to resolve this conversation if it's up to standards